x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/AutoConfigureNode.java

```diff
@@ -549,6 +573,76 @@ protected void execute(Terminal terminal, OptionSet options, Environment env) th
+            // 1. backup (move) any previously existing tls certs dir (this backup is NOT removed when auto-conf finishes)
+            if (Files.exists(env.configFile().resolve(TLS_GENERATED_CERTS_DIR_NAME))) {
+                moveDirectory(
+                    env.configFile().resolve(TLS_GENERATED_CERTS_DIR_NAME),
+                    env.configFile()
+                        .resolve(
+                            String.format(
+                                Locale.ROOT,
+                                TLS_GENERATED_CERTS_DIR_NAME + ".%d.orig",
+                                autoConfigDate.toInstant().getEpochSecond()
+                            )
+                        )
+                );
+            }
+            // 2. move the newly populated temp certs dir to its permanent static dir name
+            moveDirectory(tempGeneratedTlsCertsDir, env.configFile().resolve(TLS_GENERATED_CERTS_DIR_NAME));
+        } catch (Throwable t) {
+            // restore keystore to revert possible keystore bootstrap
+            try {
+                if (Files.exists(keystoreBackupPath)) {
+                    Files.move(
+                        keystoreBackupPath,
+                        keystorePath,
+                        StandardCopyOption.REPLACE_EXISTING,
+                        StandardCopyOption.ATOMIC_MOVE,
+                        StandardCopyOption.COPY_ATTRIBUTES
+                    );
+                } else {
+                    Files.deleteIfExists(keystorePath);
+                }
+            } catch (Exception ex) {
+                t.addSuppressed(ex);
+            }
+            // revert any previously existing TLS certs
+            try {
+                if (Files.exists(
+                    env.configFile()
+                        .resolve(
+                            String.format(
+                                Locale.ROOT,
+                                TLS_GENERATED_CERTS_DIR_NAME + ".%d.orig",
+                                autoConfigDate.toInstant().getEpochSecond()
+                            )
+                        )
+                )) {
+                    moveDirectory(
+                        env.configFile()
+                            .resolve(
+                                String.format(
+                                    Locale.ROOT,
+                                    TLS_GENERATED_CERTS_DIR_NAME + ".%d.orig",
+                                    autoConfigDate.toInstant().getEpochSecond()
+                                )
+                            ),
+                        env.configFile().resolve(TLS_GENERATED_CERTS_DIR_NAME)
+                    );
+                }
+            } catch (Exception ex) {
+                t.addSuppressed(ex);
+            }
+            try {
+                deleteDirectory(tempGeneratedTlsCertsDir);
+            } catch (Exception ex) {
+                t.addSuppressed(ex);
+            }
+            throw t;
+        }
+
```

Comment:
At this point the keystore and the certs dir are all set. What's left is to update the yml file.